preparing the new magnifier types (#19915)
nvaccess/nvda: user_docs/en/userGuide.md

The default color filter when the magnifier is first enabled can be configured in the [Magnifier settings](#MagnifierSettings).

### Magnifier Types {#MagnifierType}

The magnifier can be used in multiple modes, each designed to suit different user needs and preferences:

* Full-screen: The entire screen is magnified, and the magnified view follows the system focus or mouse pointer.

> * Fixed window: A separate window displays the magnified content, and the rest of the screen remains at normal size. This allows you to see both the magnified content and the surrounding context simultaneously.

Reviewer: ```suggestion
* Fixed window: A separate window displays the magnified content, and the rest of the screen remains at normal size.
This allows you to see both the magnified content and the surrounding context simultaneously.
```

* Docked: The magnified view is docked to one edge of the screen, providing a larger view of the area around the system focus or mouse pointer while still showing most of the screen at normal size.

> * Lens: A rectangular area around the system focus or mouse pointer is magnified, while the rest of the screen remains at normal size. This allows you to focus on a specific area without losing sight of the overall screen layout.

Reviewer: can we make sure to mention the cursor tracking here, it's the bigger differentiating factor

```suggestion
* Lens: A rectangular area around the system focus or mouse pointer is magnified, while the rest of the screen remains at normal size.
This follows the cursor around rather than staying in a fixed position.
```

Reviewer 2: @seanbudd what do you mean with "cursor"? It's often useful to clarify if we deal with text cursor (= system caret), mouse cursor (= mouse pointer), or even in our case, review cursor.

Reviewer: I think it's covered more clearly in @Boumtchack's "Lens Magnifier" section

Reviewer 2: By the way, will the lens window mode follow the same things as full screen mode?

As a point of comparison, Windows Magnifier's lens mode only follows the mouse pointer (even if can be paned away from the mouse pointer with `control+alt+arrows`). When Windows Magnifier is in lens mode, there is no way to configure what can be tracked (mouse pointer, text caret, etc.) as in full screen mode. Only the lens window size can be configured.

I'm not saying that we should do the same thing, just reflect if Windows Magnifier limitations of lens mode also apply to us for any reason or not.
In any case, I guess that lens mode is really targeting mouse users who want to manually increase a part of the screen.

I do not know if an equivalent mode exists for Zoomtext.

Author: I think we can talk about this when I'll open the Lens Magnifier PR

The default color filter when the magnifier is first enabled can be configured in the [Magnifier settings](#MagnifierSettings).

> ### Magnified Views {#MagnifiedViews}

Reviewer: I think we should avoid adding the documentation until the feature is available to users.
Instead, after this PR, let's add support and docs one at a time for each magnifier type, starting with fixed window.

The default focus mode when the magnifier is first enabled can be configured in the [Magnifier settings](#MagnifierSettings).

> #### Spotlight Mode {#MagnifierSpotlight}

Reviewer 2: We use "mode" everywhere and for all. Can we call the feature just "spotlight", not "spotlight mode" while at it?

A mode is something that seems more persistent than the spotlight feature which is a command doing something only in the next few seconds.

Reviewer 2: To avoid delaying PRs and running into merge conflicts, it may be preferable to change this in a dedicated PR. I'll open an issue for this if you agree (and may also do the PR if needed).

> #### Spotlight Mode {#MagnifierSpotlight}

Reviewer 2: I would have kept this paragraph out of the description of full screen view mode paragraph.
Even if it only applies to full screen view, it should be in a separate paragraph, described as a special feature of the Magnifier, so that the paragraph various magnifier modes (or views) focuses on describing the possible views and how they behave, not all the features that each of them allow.

Spotlight mode automatically deactivates after zooming back in.
If you move the mouse before the zoom-back occurs, the timer resets, giving you more time to view the full screen.

### Fixed Magnifier {#MagnifierFixed}

> Fixed Magnifier provides a separate window that displays a magnified view of the area around the system focus or mouse pointer. This allows you to see both the magnified content and the surrounding context simultaneously.

Reviewer: ```suggestion
Fixed Magnifier provides a separate window that displays a magnified view of the area around the system focus or mouse pointer.
This allows you to see both the magnified content and the surrounding context simultaneously.
```

> The fixed magnifier window can be resized and moved in corners, giving you flexibility in how you view magnified content.

Reviewer 2: It's not clear to me. In which corners?

Author: top left, top right, bottom left and bottom right

### Docked Magnifier {#MagnifierDocked}

> Docked Magnifier is a mode where the magnified view is docked to one edge of the screen, providing a larger view of the area around the system focus or mouse pointer while still showing most of the screen at normal size.

Reviewer 2: What is the difference of a fixed magnifier view (moved to the edges of the screen) and a docked view? Can you update the description of both so that the difference is clear?

Author: I'm starting to think that we may not need this 2 types of magnifier, maybe we should just make a combination of the 2. 
The 2 mode seems too close maybe we shouldn't go with Windows functionallity and just have 1 "docked/fixed" window that we can lock in these positions :
top left / top / top right/ right / bottom right / bottom / bottom left / left
when on a side (top/right/bottom/left) it would take all screen width, resizable height.
when on a corner (top left / top right / bottm right / bottom left ) it would be possible to resize width and height.

We could call it Windowed Magnifier. But it might confict with @CyrilleB79 idea's to change the type name to WindowedView.

Author: should we open an issue to discuss this matter? @CyrilleB79 @seanbudd

Reviewer: Let's stick to the plan of creating a fixed floating window, a docked window to an edge, and a floating window which tracks the cursor. We should start with the base fixed window, and use that as a base for the other 2

You can choose to dock the magnifier to the top, bottom, left, or right edge of the screen, depending on your preference and the layout of your applications.

### Lens Magnifier {#MagnifierLens}

> Lens Magnifier is a mode where a rectangular area around the system focus or mouse pointer is magnified, while the rest of the screen remains at normal size. This allows you to focus on a specific area without losing sight of the overall screen layout. The lens follows the system focus or mouse pointer, providing a dynamic magnification experience.

Reviewer: ```suggestion
Lens Magnifier is a mode where a rectangular area around the system focus or mouse pointer is magnified, while the rest of the screen remains at normal size.
This allows you to focus on a specific area without losing sight of the overall screen layout.
The lens follows the system focus or mouse pointer, providing a dynamic magnification experience.
```
